### ISSUE TITLE
When self.channel_name is accessed but CHANNELS_LAYER is not defined log a warning message  

### ISSUE THREAD

**Issue by pembeci** [django/channels#1667]
The docs say "Channel layers are an entirely optional part of Channels. If you don’t want to use them, just leave CHANNEL_LAYERS unset, or set it to the empty dict {}.". I defined a very basic consumer:

```python
class ChatConsumer(WebsocketConsumer):
	def connect(self):
		print(f"New USER: {self.scope['user']} {self.channel_name} ") 
		self.accept()

	def disconnect(self, close_code):
		print(f"User Left: {self.scope['user']} {self.channel_name}") 
		pass

	def receive(self, text_data):
		text_data_json = json.loads(text_data)
		message = text_data_json['message']
		print(f"data received {text_data_json}")
		self.send(text_data=json.dumps({
			'message': 'echoed: ' + message
		})) 
```
When a connection is made, I was able to see the log message in consumer's connect method but nothing else. Logs were saying `"WSCONNECTING /ws/backend/"` and then immediately `"WSDISCONNECT/ws/backend/"` and on frontend socket was opened and closed immediately.

After setting `CHANNELS_LAYER` to `"channels.layers.InMemoryChannelLayer"` it started working as expected. So if I don't have a weird local problem, it seems `CHANNELS_LAYER` can't be left undefined any more and docs should reflect that.

$ python3 -m django --version
3.1.7
$ python3 -c 'import channels; print(channels.__version__)'
3.0.3
OS: Ubuntu

**Comment by carltongibson** [comment]
Hi @pembeci — thanks for the report. I need to look into it but if it's correct then I'd say that's a bug rather than a documentation issue. We don't want to make defining the layer compulsory. (We certainly don't want people using the in memory channel layer for this!) 

Did you have a chance to dig in at all?

**Comment by pembeci** [comment]
I had an idea that I was able to test just now. My print statement includes `self.channel_name`. I thought this was always defined randomly like a socket uuid even if you don't have layers defined. When I removed it and disabled layers it worked without problems. So it was my fault partially but it would be nice if channels can throw a warning log message when `self.channel_name` is tried to be accessed but layers is not defined. Let me change the issue topic to reflect this. Feel free to close it if it is still my fault and my logging setup is not capturing such a warning message.
